test/e2e/kubectl/kubectl.go

```diff
@@ -431,6 +431,11 @@ var _ = SIGDescribe("Kubectl client", func() {
 		})
 
 		It("should support exec through kubectl proxy", func() {
+			// TODO(kubernetes/kubernetes#50466): This test
+			// fails in GKE because of the Transport layer
+			// used.
+			framework.SkipIfProviderIs("gke")
```

Comment:
it's not the provider that is the issue, it's the kubectl auth mechanism (bearer token auth will fail with the feature as it is implemented currently).